diff --git o/inst/include/Eigen/src/Core/Fill.h w/inst/include/Eigen/src/Core/Fill.h
index 779ef26..6f45935 100644
--- o/inst/include/Eigen/src/Core/Fill.h
+++ w/inst/include/Eigen/src/Core/Fill.h
@@ -93,8 +93,10 @@ struct eigen_fill_impl<Xpr, /*use_fill*/ true> {
 
 template <typename Xpr>
 struct eigen_memset_helper {
-  static constexpr bool value =
-      std::is_trivially_copyable<typename Xpr::Scalar>::value && eigen_fill_helper<Xpr>::value;
+  using Scalar = typename Xpr::Scalar;
+  static constexpr bool value = std::is_trivially_copyable<Scalar>::value &&
+                                !static_cast<bool>(NumTraits<Scalar>::RequireInitialization) &&
+                                eigen_fill_helper<Xpr>::value;
 };
 
 template <typename Xpr>
